### PR TITLE
Create midi_duration.lua

### PR DESCRIPTION
**Pull request by cv-on-hub**
New script to change playback start/stop times

### CODE REVIEW

File: src/midi_duration.lua [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    
+    if start_offset < -9999 or start_offset > 9999 or stop_offset < -9999 or stop_offset > 9999 then
+        show_error("bad_offset", start_offset .. " / " .. stop_offset)
+        return
+    end
+    if layer_number < 0 or layer_number > 4 then
+        show_error("bad_layer_number", layer_number)
+        return
+    end
+    if nil ~= finenv.RetainLuaState then
```
Comment (Nick-Mazuk) [inline src/midi_duration.lua:75]: Ok, I guess this code review is just going to be me being picky about style.

This line should be:

```lua
finenv.RetainLuaState ~= nil
```

The way you originally wrote it is called a [yoda condition](https://en.wikipedia.org/wiki/Yoda_conditions). While they are valid syntax and there are a few minor advantages of them, they can in general make it harder to read code.

**Review by Nick-Mazuk — COMMENTED** [review]
This review ended up with me just being picky about syntax, but the overall script looks good.

**Comment by cv-on-hub** [comment]
No problem with picky. The remedy is easy and I'm intrigued by your thought processes. 

I recall you saying that an `is_` test should return true/false, so I've made `is_ok` do that. I only used `RetainLuaState ~= nil` because I've seen that formulation in so many FinLua scripts and thought it was a Lua thing I didn't know about. (Compilation efficiency?) It jars my historically ingrained Perl sensibility something awful.

**Comment by Nick-Mazuk** [comment]
> I only used `RetainLuaState ~= nil` because I've seen that formulation in so many FinLua scripts and thought it was a Lua thing I didn't know about. (Compilation efficiency?)

Yeah, using `nil ~= finenv.RetainLuaState` is nothing special in Lua.

**Review by Nick-Mazuk — APPROVED** [review]
LGTM.

**Pull request merged**